src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -232,6 +273,12 @@ def on_char(self, evt):
                 self._shape_to_copy = self._selected_shape
                 self.cnvs.set_default_cursor(wx.CURSOR_BULLSEYE)
                 self.cnvs.request_drawing_update()
+            elif evt.GetKeyCode() == wx.WXK_CONTROL_Z:
+                self._selected_shape.undo()
+                self._selected_shape.refresh()
```

Comment:
I'd argue the `refresh()` should be done automatically by the `.restore_state()`.